tests/unit/test_base.py

```diff
@@ -201,3 +201,25 @@ def test_inequality_no_id(self, fake_manager):
         obj1 = FakeObject(fake_manager, {"attr1": "foo"})
         obj2 = FakeObject(fake_manager, {"attr1": "bar"})
         assert obj1 != obj2
+
+    def test_dunder_str(self, fake_manager):
+        obj1 = FakeObject(fake_manager, {"attr1": "foo"})
+        assert str(obj1) == (
+            "<class 'tests.unit.test_base.FakeObject'> => {'attr1': 'foo'}"
+        )
+
+    def test_pformat(self, fake_manager):
+        obj1 = FakeObject(fake_manager, {"attr1": "foo"})
+        assert obj1.pformat() == (
+            "<class 'tests.unit.test_base.FakeObject'> => \n{'attr1': 'foo'}"
+        )
+
+    def test_pprint(self, capfd, fake_manager):
+        obj1 = FakeObject(fake_manager, {"attr1": "foo"})
+        result = obj1.pprint()
+        assert result is None
+        stdout, stderr = capfd.readouterr()
+        assert stdout == (
+            "<class 'tests.unit.test_base.FakeObject'> => \n{'attr1': 'foo'}\n"
+        )
+        assert stderr == ""
```

Comment:
I know we'll have to change it elsewhere later as well but I think you might like this suggestion to `fake_object` :grin:
```suggestion
    def test_dunder_str(self, fake_manager):
        fake_object = FakeObject(fake_manager, {"attr1": "foo"})
        assert str(fake_object) == (
            "<class 'tests.unit.test_base.FakeObject'> => {'attr1': 'foo'}"
        )

    def test_pformat(self, fake_manager):
        fake_object = FakeObject(fake_manager, {"attr1": "foo"})
        assert fake_object.pformat() == (
            "<class 'tests.unit.test_base.FakeObject'> => \n{'attr1': 'foo'}"
        )

    def test_pprint(self, capfd, fake_manager):
        fake_object = FakeObject(fake_manager, {"attr1": "foo"})
        result = fake_object.pprint()
        assert result is None
        stdout, stderr = capfd.readouterr()
        assert stdout == (
            "<class 'tests.unit.test_base.FakeObject'> => \n{'attr1': 'foo'}\n"
        )
        assert stderr == ""
```

Also was thinking if we should do `FakeObject(fake_manager, {"attr1": "foo", "attr2": "bar"})` to show it pretty-prints dicts with new-lines but not sure it's worth it. Up to you!

Reply:
> I know we'll have to change it elsewhere later as well but I think you might like this suggestion to `fake_object` 😁

Works for me 😊

> Also was thinking if we should do `FakeObject(fake_manager, {"attr1": "foo", "attr2": "bar"})` to show it pretty-prints dicts with new-lines but not sure it's worth it. Up to you!

Done!